sway/input/input-manager.c

```diff
@@ -0,0 +1,315 @@
+#define _XOPEN_SOURCE 700
+#include <ctype.h>
+#include <float.h>
+#include <limits.h>
+#include <stdio.h>
+#include <string.h>
+#include <libinput.h>
+#include <math.h>
+#include "sway/config.h"
+#include "sway/input/input-manager.h"
+#include "sway/input/seat.h"
+#include "sway/server.h"
+#include "stringop.h"
+#include "list.h"
+#include "log.h"
+
+static const char *default_seat = "seat0";
+
+// TODO make me not global
+struct sway_input_manager *input_manager;
+
+struct input_config *current_input_config = NULL;
+struct seat_config *current_seat_config = NULL;
+
+static struct sway_seat *input_manager_get_seat(
+		struct sway_input_manager *input, const char *seat_name) {
+	struct sway_seat *seat = NULL;
+	wl_list_for_each(seat, &input->seats, link) {
+		if (strcmp(seat->wlr_seat->name, seat_name) == 0) {
+			return seat;
+		}
+	}
+
+	return sway_seat_create(input, seat_name);
+}
+
+static inline int strlen_num(int num) {
+	if (num == 0) {
+		return 2;
+	}
+	return (int)((ceil(log10(abs(num)))+2));
+}
+
+static char *get_device_identifier(struct wlr_input_device *device) {
+	int vendor = device->vendor;
+	int product = device->product;
+	char *name = strdup(device->name);
+	name = strip_whitespace(name);
+
+	char *p = name;
+	for (; *p; ++p) {
+		if (*p == ' ') {
+			*p = '_';
+		}
+	}
+
+	int len =
+		(strlen(name) +
+		strlen_num(device->vendor) +
+		strlen_num(device->product) +
+		3) * sizeof(char);
```

Comment:
Can you just use `int len = snprintf(NULL, 0, ...)` here

Reply:
+1